### PR TITLE
Numpad Keys

### PR DESCRIPTION
**Pull request by shimmytoast**
Added numpad keys

### CODE REVIEW

File: key_map.h [hunk 1]
```diff
@@ -1,362 +1,379 @@
...
+// Be aware. The enum below doesn't include keys like '!' and '{' because they
+// are shift-'1' and shift-'[' respectively.
+enum KeyCode : unsigned char {
+  bks = 8,   // backspace
+  tab = 9,   // tab
+  ent = 13,  // enter
+  esc = 27,  // escape
+  spc = 32,  // or just ' '
+  
+  pls = 199, // keypad plus
```
Comment (macroxue) [inline key_map.h:18]: Can you please move this new section down so the enum values are sorted.

Also I'm not sure why this change shows up as a full file modification rather than just added lines. It's really hard to see what exact changes are made. Do you mind submitting a pull request with only the diffs?

File: keyboard-firmware.ino [hunk 2]
```diff
@@ -1,42 +1,43 @@
...
+// Include one and only one layout below.
+//#include "layouts/left_split.h"
+//#include "layouts/right_split.h"
+//#include "layouts/geng_yun.h"
+//#include "layouts/geng_yun_vi.h"
+//#include "layouts/so_ho.h"
+//#include "layouts/so_ho_mini.h"
+//#include "layouts/darknight.h"
+//#include "layouts/darknight2.h"
+#include "layouts/compmech.h"
```
Comment (macroxue) [inline keyboard-firmware.ino:17]: This won't work well when we have more than one user of this firmware. Let me see if there is a better way, like each layout gets its own directory and makefile.

File: layouts/compmech.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+constexpr int R = 5, C = 19;
+
+// The row pins, to be modified according to the exact wiring.
+const int row_pins[R] = { 20, 21, 22, 23, 24 };
+
+// The column pins, to be modified according to the exact wiring.
+const int col_pins[C] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19};
+
+Layer<R,C> layers[] = {
+  { "qwerty", l0, l1, "",
```
Comment (macroxue) [inline layouts/compmech.h:11]: Since you don't have a 'fn' key, this line can be simplified to "qwerty", l0, l0, "".

File: layouts/compmech.h [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    {
+      {   f1,  f2, esc,  '1',  '2',  '3',  '4',  '5',  '6',  '7',  '8',  '9',  '0',   '-',   '=',   bks,  nml,  scl,  kmn },
+      {   f3,  f4, tab,  'q',  'w',  'e',  'r',  't',  'y',  'u',  'i',  'o',  'p',   '[',   ']',   ent,  kn7,  kn8,  kn9 },
+      {   f5,  f6, cap,  'a',  's',  'd',  'f',  'g',  'h',  'j',  'k',  'l',  ';',  '\'',   '`',   kmt,  kn4,  kn5,  kn6 },
+      {   f7,  f8, lal, '\\',  'z',  'x',  'c',  'v',  'b',  'n',  'm',  ',',  '.',   '/',   rct,   kdl,  kn1,  kn2,  kn3 },
+      {   f9, f10, lsh ,  0 ,   0 ,   0 ,   0 ,   0 ,  spc,   0 ,   0 ,   0 ,   0 ,    0 ,   rsh,   0,    0,    kn0,  pls },
+    }
+  },
+  { "fn-media", l1, l1, "",
```
Comment (macroxue) [inline layouts/compmech.h:20]: This layer looks the same as l0. Since there is no way to get into this layer, it can be deleted.

**Review by macroxue — COMMENTED** [review]
This is really cool! It's great to know that there is another user of the firmware out there. I left a few comments in the code. Please see if they make sense. Thanks!

File: keyboard-firmware.ino (hunk 2 above)
Comment (macroxue) [reply keyboard-firmware.ino:17]: It's fixed by 71b3986 so each keyboard has its own directory now. The only slight inconvenience is that a symbolic link needs to be created in Arduino libraries directory first. Please check the updated README.

File: key_map.h (hunk 1 above)
Comment (macroxue) [reply key_map.h:18]: I added keypad keys with 90b92c0 and credited you in the commit. Hopefully you don't mind.

There were some name changes for consistency and two additional keys: KEYPAD_SLASH and KEYPAD_ENTER.

**Comment by shimmytoast** [comment]
Hey thanks for the updates and all sorry I don't reply in a timely manner. I have no clue what I'm doing with coding so anything I did was more or less a hack lol. Thanks for all the tips and for adding the support for the keys proper!
